### PR TITLE
Strong type `Dependabot::BranchNamer::DependencyGroupStrategy`

### PR DESCRIPTION
**Pull request by JamieMagee**

### CODE REVIEW

File: common/lib/dependabot/pull_request_creator/branch_namer.rb [hunk 1]
```diff
@@ -49,10 +49,10 @@ def strategy
               files: files,
               target_branch: target_branch,
               dependency_group: dependency_group,
+              includes_security_fixes: includes_security_fixes,
```
Comment (JamieMagee) [inline common/lib/dependabot/pull_request_creator/branch_namer.rb:52]: Sorbet needs required properties to come before optional properties.

File: common/lib/dependabot/pull_request_creator/branch_namer.rb [hunk 2]
```diff
@@ -49,10 +49,10 @@ def strategy
...
               separator: separator,
               prefix: prefix,
-              max_length: max_length,
-              includes_security_fixes: includes_security_fixes
```
Comment (abdulapopoola) [inline common/lib/dependabot/pull_request_creator/branch_namer.rb:55]: any reason for the reordering?

Comment (JamieMagee) [reply common/lib/dependabot/pull_request_creator/branch_namer.rb:55]: Sorbet needs required properties to come before optional properties. `includes_security_fixes` is required, but doesn't have a default value.

File: common/spec/dependabot/pull_request_creator/branch_namer/dependency_group_strategy_spec.rb [hunk 3]
```diff
@@ -130,7 +130,7 @@
 
     context "with a grouped security update" do
       let(:directory) { "/" }
-      let(:target_branch) { nil }
+      let(:target_branch) { "" }
```
Comment (abdulapopoola) [inline common/spec/dependabot/pull_request_creator/branch_namer/dependency_group_strategy_spec.rb:133]: nice!

**Review by abdulapopoola — APPROVED** [review]

**Review by bdragon — APPROVED** [review]

File: common/spec/dependabot/pull_request_creator/branch_namer/dependency_group_strategy_spec.rb (hunk 3 above)
Comment (JamieMagee) [reply common/spec/dependabot/pull_request_creator/branch_namer/dependency_group_strategy_spec.rb:133]: I think in practice this is always a `String`, but a lot of tests take shortcuts, which meant it was `T.nilable(String)`.

**Pull request merged**